RecoTracker/TkDetLayers/src/Phase2EndcapSubDisk.cc

```diff
@@ -0,0 +1,85 @@
+#include "Phase2EndcapSubDisk.h"
+
+#include "FWCore/MessageLogger/interface/MessageLogger.h"
+
+#include "DataFormats/GeometrySurface/interface/SimpleDiskBounds.h"
+
+#include "TrackingTools/DetLayers/interface/DetLayerException.h"
+#include "TrackingTools/GeomPropagators/interface/HelixForwardPlaneCrossing.h"
+
+#include <array>
+#include "DetGroupMerger.h"
+
+using namespace std;
+
+//hopefully is never called!
+const std::vector<const GeometricSearchDet*>& Phase2EndcapSubDisk::components() const {
```

Comment:
I think it's difficult to move this one since its argument would be something that could be a nullptr (hence the check). I'm also not sure why it's needed (I assumed I'd need it here since it's in the already existing class so presumably there is some edge case where this function could get called.) I think it's not really necessary but to avoid any potential seg faults I could just change this function to throw an error in case it gets called. Would that make sense?

Reply:
do I understand correctly that the implementation of this function is just a copy from the old classes and hasn't been explicitly tested?

If so, I think throwing `throw cms::Exception("Not Implemented")` here instead might be more appropriate, as otherwise we have no idea if the implementation is relevant or correct.

Reply:
Yes, you are correct. I've changed it to throw an exception